app/models/user.rb

```diff
@@ -1,5 +1,13 @@
 class User < ApplicationRecord
   has_many :microposts, dependent: :destroy
+  has_many :active_relationships, class_name:  "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent:   :destroy
+  has_many :following, through: :active_relationships, source: :followed
```

Comment:
việc khai báo association `through` có tác dụng gì thế em?

Reply:
Việc dùng `through` có tác dụng tránh việc dư thừa dữ liệu. Vì khi không `through` qua `active_relationships` thì mỗi relationship của 1 user ta phải lưu cả user_id và những thông tin liên quan của user đó

Reply:
Vẫn chưa cụ thể được tác dụng của nó nhé @@

Reply:
`through` sẽ tạo nên kết nối `many-to-many` giữa các model. Vì 1 user có thể follow nhiều user khác, và cũng có thể được follow bởi nhiều user

Reply:
@@, cụ thể là khi khai báo `through` kết nối many-to-many giữa các model thì có thể kết nối trực tiếp đến 2 model parent mà không phải thông qua model child nhé
VD: 
```ruby
@user = User.first

user_following = @user.following
user_followers = @user.followers
```